cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/ConsumerConfiguration.java

```diff
@@ -0,0 +1,128 @@
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset.lib.partitioned;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.dataset.lib.PartitionDetail;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Configuration parameters to be used by a {@link PartitionConsumer}.
+ */
+public class ConsumerConfiguration {
+  /**
+   * Instance of ConsumerConfiguration with default parameters
+   */
+  public static final ConsumerConfiguration DEFAULT = builder().build();
+
+  private final Predicate<PartitionDetail> partitionPredicate;
+  private final int maxWorkingSetSize;
+  private final long timeout;
+
+  private ConsumerConfiguration(Predicate<PartitionDetail> partitionPredicate,
+                                int maxWorkingSetSize, long timeout) {
+    this.partitionPredicate = partitionPredicate;
+    this.maxWorkingSetSize = maxWorkingSetSize;
+    this.timeout = timeout;
+  }
+
+  /**
+   * @return A predicate to be applied on {@link PartitionDetail}s to determine which partitions to include in the
+   *         partition consumption.
+   */
+  public Predicate<PartitionDetail> getPartitionPredicate() {
+    return partitionPredicate;
+  }
+
+  /**
+   * @return An upper bound on the size of the working set of partitions that get serialized as part of the consumer's
+   *         state.
+   */
+  public int getMaxWorkingSetSize() {
+    return maxWorkingSetSize;
+  }
+
+  /**
+   * Defines an expiration timeout, in seconds, of IN_PROGRESS partitions
+   * @return number of seconds that a partition can be in progress before it is deemed failed.
+   */
+  public long getTimeout() {
+    return timeout;
+  }
+
+  /**
+   * @return a {@link Builder} instance to build an instance of a ConsumerConfiguration.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * A Builder to construct ConsumerConfiguration instances.
+   */
+  public static class Builder {
+    /**
+     * Default values
+     */
+    private Predicate<PartitionDetail> partitionPredicate = new Predicate<PartitionDetail>() {
+      @Override
+      public boolean apply(@Nullable PartitionDetail input) {
+        return true;
+      }
+    };
+    private int maxWorkingSetSize = 1000;
+    // 12 hour timeout
+    private long timeout = TimeUnit.HOURS.toSeconds(12);;
+
+    private Builder() {
+    }
+
+    /**
+     * Sets the Predicate of the ConsumerConfiguration.
+     */
+    public Builder setPartitionPredicate(Predicate<PartitionDetail> partitionPredicate) {
+      this.partitionPredicate = partitionPredicate;
+      return this;
+    }
+
+    /**
+     * Sets the maximum working set size of the ConsumerConfiguration.
+     */
+    public Builder setMaxWorkingSetSize(int maxWorkingSetSize) {
+      this.maxWorkingSetSize = maxWorkingSetSize;
+      return this;
+    }
+
+    /**
+     * Sets the timeout of the ConsumerConfiguration.
+     */
+    public Builder setTimeout(long timeout) {
```

Comment:
It is not clear to me what the timeout means. If a partition in marked as in-progress, and it times out, what happens? Is it considered new again? Or is it discarded from the working set? Is that documented somewhere?